### ISSUE TITLE
delivery.license as created by chef-ingredient 

### ISSUE THREAD

**Issue by mmccollu** [chef-cookbooks/chef-ingredient#195]
### Chef-client version
chef-client version: 12.21.2

### Platform Details
SLESsp2

### Scenario:
The owner and permissions of delivery.license as created by chef-ingredient do not match those as when the command "automate-ctl reconfigure" is run. This is causing the license to be unreadable and shows up as expired

### Steps to Reproduce:
chef-val-wfl.hplab1.ford.com cookbook_file[/opt/delivery/embedded/cookbooks/delivery/recipes/nginx.rb] action create[2017-08-07T09:43:56-04:00] INFO: Processing cookbook_file[/opt/delivery/embedded/cookbooks/delivery/recipes/nginx.rb] action create (ford_pipeline_build::chef_automate line 47)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com chef_automate[chef-val-wfl.hplab1.ford.com] action create[2017-08-07T09:43:56-04:00] INFO: Processing chef_automate[chef-val-wfl.hplab1.ford.com] action create (ford_pipeline_build::chef_automate line 56)
chef-val-wfl.hplab1.ford.com
chef-val-wfl.hplab1.ford.com chef_ingredient[automate] action upgrade[2017-08-07T09:43:56-04:00] INFO: Processing chef_ingredient[automate] action upgrade (/var/chef/cache/cookbooks/chef-ingredient/resources/automate.rb line 51)
chef-val-wfl.hplab1.ford.com
chef-val-wfl.hplab1.ford.com ruby_block[stop chef run] action nothing[2017-08-07T09:43:56-04:00] INFO: Processing ruby_block[stop chef run] action nothing (/var/chef/cache/cookbooks/chef-ingredient/libraries/helpers.rb line 251)
chef-val-wfl.hplab1.ford.com (skipped due to action :nothing)
chef-val-wfl.hplab1.ford.com remote_file[/var/chef/cache/automate-1.5.46-1.el6.x86_64.rpm] action create[2017-08-07T09:43:56-04:00] INFO: Processing remote_file[/var/chef/cache/automate-1.5.46-1.el6.x86_64.rpm] action create (/var/chef/cache/cookbooks/chef-ingredient/libraries/default_handler.rb line 104)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com zypper_package[automate] action upgrade[2017-08-07T09:43:58-04:00] INFO: Processing zypper_package[automate] action upgrade (/var/chef/cache/cookbooks/chef-ingredient/libraries/default_handler.rb line 55)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com directory[/etc/delivery] action create[2017-08-07T09:43:58-04:00] INFO: Processing directory[/etc/delivery] action create (/var/chef/cache/cookbooks/chef-ingredient/resources/automate.rb line 62)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com directory[/etc/chef] action create[2017-08-07T09:43:58-04:00] INFO: Processing directory[/etc/chef] action create (/var/chef/cache/cookbooks/chef-ingredient/resources/automate.rb line 62)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com directory[/var/opt/delivery/license] action create[2017-08-07T09:43:58-04:00] INFO: Processing directory[/var/opt/delivery/license] action create (/var/chef/cache/cookbooks/chef-ingredient/resources/automate.rb line 62)
chef-val-wfl.hplab1.ford.com (up to date)
chef-val-wfl.hplab1.ford.com chef_file[/var/opt/delivery/license/delivery.license] action create[2017-08-07T09:43:58-04:00] INFO: Processing chef_file[/var/opt/delivery/license/delivery.license] action create (/var/chef/cache/cookbooks/chef-ingredient/resources/automate.rb line 73)
chef-val-wfl.hplab1.ford.com
chef-val-wfl.hplab1.ford.com * cookbook_file[/var/opt/delivery/license/delivery.license] action create[2017-08-07T09:43:58-04:00] INFO: Processing cookbook_file[/var/opt/delivery/license/delivery.license] action create (/var/chef/cache/cookbooks/chef-ingredient/resources/file.rb line 40)
chef-val-wfl.hplab1.ford.com [2017-08-07T09:43:58-04:00] INFO: cookbook_file[/var/opt/delivery/license/delivery.license] owner changed to 0
chef-val-wfl.hplab1.ford.com [2017-08-07T09:43:58-04:00] INFO: cookbook_file[/var/opt/delivery/license/delivery.license] mode changed to 600
chef-val-wfl.hplab1.ford.com
chef-val-wfl.hplab1.ford.com - change mode from '0644' to '0600'
chef-val-wfl.hplab1.ford.com - change owner from 'deliver5' to 'root'
chef-val-wfl.hplab1.ford.com

### Expected Result:
[What are you expecting to happen as the consequence of above reproduction steps?]

### Actual Result:
[What actually happens after the reproduction steps? Include the error output or a link to a gist if possible.]

**Comment by ncerny** [comment]
Hey @mmccollu - what are you stating the permissions should be?  Also, can you confirm that it works if you have the correct permissions?

Would you be able to submit a pull request to resolve this?

You can find me in Slack if you have any questions or need help.

**Comment by mmccollu** [comment]
Nathan I have opened a ticket with chef #16009.
Told to create an issue on the chef ingredient github repo, this was after I submitted the ticket on CHEF support and it would be sent to chef/dev.

Thanks Mike

**Comment by mmccollu** [comment]
I think we need to clarify the issue; the problem is that when the installer.sh runs the first time everything works normally. However, on subsequent runs to modify an existing environment the automate-ctl reconfigure does not run which leaves the delivery.license with the wrong permissions with the end result being the license shows up as expired.

From: McCollum, Michael (M.J.)
Sent: Wednesday, August 16, 2017 10:44 AM
To: chef-cookbooks/chef-ingredient <reply@reply.github.com>
Cc: Sean Horn (Chef) <support@getchef.zendesk.com>; Cardwell, Kerry (K.L.) <kcardwel@ford.com>; Bodnar, Joseph (J.S.) <jbodnar@ford.com>
Subject: RE: [chef-cookbooks/chef-ingredient] delivery.license as created by chef-ingredient (#195)

Nathan I have opened a ticket with chef #16009.
Told to create an issue on the chef ingredient github repo, this was after I submitted the ticket on CHEF support and it would be sent to chef/dev.

Thanks Mike
